### ISSUE TITLE
Lack of documentation on using amplify-authenticator in an Angular App

### ISSUE THREAD

**Issue by Rajan-Gupta1** [aws-amplify/amplify-ui#2314]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have tried disabling all browser extensions or using a different browser
- [X] I have tried deleting the node_modules folder and reinstalling my dependencies
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

Angular

### Which UI component?

Authenticator

### How is your app built?

Angular

### What browsers are you seeing the problem on?

Chrome

### Please describe your bug.

I am wanting to integrating amplify-authenticator in am open source app https://github.com/changhuixu/angular-material-admin as an exercise before I integrate it in production app. I am able to call the onLogin() method which I think is a crude way of doing it and the navigation  to protected route 'dashboard' work. but then when the user clicks logout (explained below) and is sent to [/login] route, the onLogin() method goes into an infinite loop. I am seeking help in how to professionally integrate login and logout functionality in an application such as https://angular-material-admin.firebaseapp.com/
If someone from the amplify team can integrate the Amplify Auth functionaluty in this demo app, I think it will be a great service to amplify users. I am also not able to find good documentation similar to react for protected routes
Below is the code for login.component.html 

```html
<amplify-authenticator [loginMechanisms]="['email']" [hideSignUp]="false">
  <ng-template amplifySlot="authenticated"
                let-user='user'
                let-signOut='signOut'
                let-route='route'
                let-authStatus='authStatus'>
  {{ onLogin() }}
</ng-template>
</amplify-authenticator>
```
and here is the code for onLogin() method in login.component.ts
```ts
  onLogin() {
    Auth.currentAuthenticatedUser({
      bypassCache: false
    })
    .then((user) => {
      console.log(user);
      var cognitoUser :CognitoUser = user;
      if (user) {
        this.router.navigate(['/dashboard']);
      }
    })
  }
```

My file top-nav,component.ts which has the logout button invokes the onLoggedOut() method.

```ts
 async onLoggedout() {
    console.log("TopNavComponent::onLoggedout");

    await Amplify.Auth.signOut();
      this.router.navigate(['/login']);
  }
```

### What's the expected behaviour?

After the user logged out, the system goes into an infinite loop

### Help us reproduce the bug!

```html
<amplify-authenticator [loginMechanisms]="['email']" [hideSignUp]="false">
  <ng-template amplifySlot="authenticated"
                let-user='user'
                let-signOut='signOut'
                let-route='route'
                let-authStatus='authStatus'>
  {{ onLogin() }}
</ng-template>
</amplify-authenticator>
```

**Issue 1: onLogin method is being called twice**
**Issue 2: What is the correct way of using amplify-auth in a professional angular app with route protection and log out** functionality

### Code Snippet

```typescript
// Put your code below this line.

```

\* *edited by wlee221 for code formatting*

**Comment by wlee221** [comment]
Sounds good! I'll try that.

**Comment by wlee221** [comment]
I got it running! Please check https://github.com/wlee221/angular-material-admin-amplify for my code.

This [commit](https://github.com/wlee221/angular-material-admin-amplify/commit/b0483bb306b3d93fb5551a3dc58971372acd2608) shows how I integrated Amplify. It's worth noting that I used one private APIs to make this work (see [comment](https://github.com/wlee221/angular-material-admin-amplify/commit/b0483bb306b3d93fb5551a3dc58971372acd2608#r79031259)). We'll work on making this API official so it feels less hacky. 

But this should be working, and I hope this unblocks you!

**Comment by wlee221** [comment]
As per official guide, we'll write a more formal guide like we did to React. Just wanted to unblock while we take the time to do so 🙏

**Comment by Rajan-Gupta1** [comment]
@wlee221  I have added this and it seems to work. Many Thanks...
1. Suggestion, in angular typically a service is @injectable and requires to be declared in the providers of a module. Calling AuthenticatorService as a service in its name will likely confuse angular programs and made them think its a service and should be declared in the provider. See reference to angular doc here https://angular.io/guide/providers
2. Instead of having to use a deprecated method of subscribe, is there another pattern that I can use today to build my angular app? If not possible, how would I be notified that a long term solution has been found for this workaround so that I can update my code?

**Comment by wlee221** [comment]
1. You're right! Let me make the right adjustment. `AuthenticatorService` is an injectable though: https://github.com/aws-amplify/amplify-ui/blob/1ad2a4cc445e66c7f7b39b648e2b02e6a97bbcd1/packages/angular/projects/ui-angular/src/lib/services/authenticator.service.ts#L22-L25

so it might be just matter of updating our docs to match Angular conventions.

2. To be honest, we do not have better pattern to handle that. What we can do is to create a new feature request GitHub issue in this repo, and you can use that issue to subscribe and track it. Until that issue is resolved, I would stick to `subscribe` method and we'll be careful not to change that.

**Comment by Rajan-Gupta1** [comment]
My bad as I was looking at the compiled code which was as below. So how come angular did not give me an error since I did not put this in providers for the module. 
/**
 * AuthenticatorService provides access to the authenticator state and context.
 */
import * as ɵngcc0 from '@angular/core';
export declare class AuthenticatorService implements OnDestroy {
    private _authState;
    private _authService;
    private _sendEventAliases;
    private _machineSubscription;
    private _facade;
    constructor();

**Comment by Rajan-Gupta1** [comment]
@wlee221  as requested in [https://github.com/aws-amplify/amplify-ui/issues/2314#issuecomment-1191924995](https://github.com/aws-amplify/amplify-ui/issues/2314#issuecomment-1191924995) a new feature request has been created with reference [https://github.com/aws-amplify/amplify-ui/issues/2324#issue-1313864256](https://github.com/aws-amplify/amplify-ui/issues/2324#issue-1313864256)

**Comment by Rajan-Gupta1** [comment]
Marking this issue closed with the workaround and provided angular pattern. Will test this out in more detail in my app for now closing
